File: Arrays/Count_Triplets.cpp
class Solution{
public:	
	
	int countTriplet(int arr[], int n)
	{
	    // Your code goes here
	    sort(arr,arr + n);
	    int ans =0;
	    for(int i = n-1;i>=2;i--){
	    int start =0,end = i-1;
	    while(start<end){
	        if(arr[start]+arr[end]==arr[i]){ans++;start++;end--;}
	        else if(arr[start]+arr[end]<arr[i]){start++;}
	        else {end--;}
	    }
	    }
	    return ans;
	}
};

File: Arrays/Count_Triplets.py
class Solution:
	def countTriplet(self, arr, n):
		# code here
		arr.sort()
		ans = 0
		for i in range(n-1,1,-1):
		    start,end = 0,i-1
		    while(start<end):
		        sum = arr[start]+arr[end]
		        if sum == arr[i]:
		            ans+=1
		            start+=1
		            end-=1
		        elif sum < arr[i]:
		            start+=1
		        else:
		            end-=1
        return ans
